src/adapters/mc/OsConfigResource.c

```diff
@@ -656,6 +681,17 @@ void MI_CALL OsConfigResource_Invoke_GetTargetResource(
         }
     }
 
+    // Read the procedure MIM object value from the input resource values
+    if ((in->InputResource.value->ProcedureObjectValue.exists == MI_TRUE) && (in->InputResource.value->ProcedureObjectValue.value != NULL))
+    {
+        FREE_MEMORY(g_procedureObjectValue);
+        if (NULL == (g_procedureObjectValue = DuplicateString(in->InputResource.value->ProcedureObjectValue.value)))
+        {
+            LogError(context, miResult, GetLog(), "[OsConfigResource.Get] DuplicateString(%s) failed", in->InputResource.value->ProcedureObjectValue.value);
+            g_procedureObjectValue = DuplicateString(g_defaultValue);
```

Comment:
Is no `goto Exit;` here intentional?

Reply:
Correct. The new 'procedure' parameters are optional, they may not always exist in the MOF, and we must not fail when not.